docs/adr/2023-07-05_024-offline-mode.md

```diff
@@ -0,0 +1,32 @@
+---
+slug: 25
+title: | 
+  1. Offline mode
+authors: [cardenaso11]
+tags: []
+---
+## Status
+Proposed
+
+## Context
+Currently, the Hydra node requires a Layer 1 Cardano node running in order to operate; The L1 node is needed to submit and watch for L1 transactions. Generally speaking, the transactions watched are for learning the state of peer Hydra nodes. The transactions submitted are to transition between states (e.g. after submitting a Commit tx to the L1, a node watches to see when all other nodes have also Comitted.)
+
+There are applications for the Hydra node where interaction with a Layer 1 chain is unnecessary. Sundaeswap is developing the Gummiworm protocol, a Layer 2 rollup chain for Cardano. Gummiworm intends to have dedicated "validator node" entities, that validate & challenge the full history of transactions for a given Layer 2 Gummiworm session. These validator nodes don't need to connect to any peers, and each will be driven by a single client. SundaeSwap intends to use modified Hydra nodes, running in offline mode, to act as these validator nodes.
+
+The Hydra node offline mode would remove the dependency on the L1 Cardano node, for applications like Gummiworm where it is unneeded. This would be useful for other Layer 2s that build on top of Hydra instead of duplicating its efforts, and for anyone who wants to easily validate a set of Cardano transactions.
+## Decision
+Hydra node will be executable in offline mode, as an alternative to the default online mode. When online, the Hydra node depends on querying a Cardano node for Era History information and Genesis parameters. When offline this is not necessary, because the Hydra node will not connect to any Layer 1.
+
+The initial state of the head will be specified in a flag, which makes any Commit redundant. The flag will specify a file for the starting Layer 2 UTXO. The Hydra node can be configured to wrte the current UTXO into a file, including the starting UTXO file.
+
+A node running in offline mode will not be able to switch between offline and online modes once started, as it is an unlikely use-case that would likely add more complexity.
+
+Commit endpoint will return 400 instead of building a transaction, in offline mode.
+
+<!-- NOTE: do we want to support having peer hydra nodes in offline/mock chain mode? Would need to add networking to replace communication done over L1 -->
```

Comment:
We should decide on this `NOTE`. The simplest solution is to say the node will run without any network connection to other peers, which will also improve performance following https://github.com/input-output-hk/hydra/pull/1101

Reply:
Agree on this. After thinking about it for a bit, it's something that could be added later if there's a usecase for it, but I think currently it doesn't seem important enough to block on it